### ISSUE TITLE
Data Manager: Update the Project Structure

### ISSUE THREAD

**Issue by VincenzoFerme** [benchflow/benchflow#293]
@jeppe-style link to the standard
@igormoreno comply with the standard

In the process also the gitignore has to be updated so that we keep locally to the project, only ignored files that are local to the project, and we move in the gitignore in the root the rest.

**Comment by jeppe-style** [comment]
I suggest that we follow the API standard as agreed for the test-manager and experiment-manager by resource based. See https://github.com/benchflow/benchflow/tree/devel/benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/resources

**Comment by igormoreno** [comment]
What's the standard? Where is it?

**Comment by jeppe-style** [comment]
basically it is as described in the resources link above:
e.g. for experiment: `/v1/users/{username}/tests/{testName}/{testNumber}/experiments/{experimentNumber}/`

I use this to convert from an experimentID to an API url https://github.com/benchflow/benchflow/blob/devel/benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/constants/BenchFlowConstants.java#L91

**Comment by VincenzoFerme** [comment]
@jeppe-style could you please describe also how the expected project structure is, as well as how to structure the tests using the scaffolding we defined for the `test-manager`? I think this is also missing in the `data-manager`

**Comment by jeppe-style** [comment]
The project structure is based on the [dropwizard suggestions](http://www.dropwizard.io/1.1.0/docs/manual/core.html)

- `api` - request, response objects (it is now part of the application project)
- `client` - client to interact with service (currently not implemented)
- `configurations` - everything related to the application configuration (`configuration.yml`)
- `constants` - (temp) constants that are cross service
- `demo` - (temp) work around solutions needed to run with currently deployed version of BenchFlow
- `exceptions` - exceptions
- `models` - DB models
- `resources` - the different REST resources
- `services` - classes that interacts with services (e.g. DB, mongo, experiment manager)

Then there are also some project specific packages for the logic of the service.

For testing (I think I wrote this also somewhere else but could not find it back):
- `*Test` - any file that contains jUnit tests (or similar) that are run during `mvn test`
- `*IT` - integration test - tests that call/starts services (including the application itself)

To run the integration tests with docker dependencies we have created a `DockerComposeIT` [class](https://github.com/benchflow/benchflow/blob/devel/benchflow-test-manager/application/src/test/java/cloud/benchflow/testmanager/DockerComposeIT.java). By specifying a docker compose file with the dependent services we can start their containers and interact with them. Example [docker-compose.yml](https://github.com/benchflow/benchflow/tree/devel/benchflow-test-manager/application/src/test/resources/docker-compose)

**Comment by igormoreno** [comment]
I think we need to talk about this. I mean it's probably more productive in a meeting

**Comment by igormoreno** [comment]
This issue refers only to the folder structure.
- Issue referring to URL standard:  #349 
- Issue referring to integration tests: #348

**Comment by igormoreno** [comment]
From @jeppe-style:
- Why do we have the package `service`? We don't have that in the other services.

**Comment by igormoreno** [comment]
@jeppe-style you wrote about the folder structure that inside the folder configurations:

> configurations - everything related to the application configuration (configuration.yml)

but I think looking at the other projects this folder should contain only Java related configuration files but the `configuration.yml` file seems to be In the root of the project. Is that correct?

**Comment by jeppe-style** [comment]
@igormoreno yes, the package `configuration` contains the classes that the `configuration.yml` (in the project root) will be read into
